src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/movements/MovementsService.kt

```diff
@@ -30,39 +29,43 @@ class MovementsService(
       throw NotFoundException("Offender with nomsId=$offenderNo not found")
     }
 
-    val bookings = offenderBookingRepository.findAllByOffenderNomsId(offenderNo)
-
-    bookings.fixMergedSchedules()
+    val movementApplications = offenderMovementApplicationRepository.findAllByOffenderBooking_Offender_NomsId(offenderNo)
+      .also { it.fixMergedSchedules() }
+    val unscheduledTemporaryAbsences = temporaryAbsenceRepository.findAllByOffenderBooking_Offender_NomsIdAndScheduledTemporaryAbsenceIsNull(offenderNo)
+    val unscheduledTemporaryAbsenceReturns = temporaryAbsenceReturnRepository.findAllByOffenderBooking_Offender_NomsIdAndScheduledTemporaryAbsenceIsNull(offenderNo)
```

Comment:
The main change is to drive the retrieval of applications and unscheduled movements from the offenderNo rather than by getting all bookings as this means there is a single query for each - previously we had n+1 queries on each.